satellite/satellitedb/overlaycache.go

```diff
@@ -22,6 +24,251 @@ type overlaycache struct {
+			&overlayNode.AuditCount, &overlayNode.AuditSuccessCount,
+			&overlayNode.UptimeCount, &overlayNode.UptimeSuccessCount)
+		if err != nil {
+			return nil, err
+		}
+
+		node, err := convertOverlayNode(overlayNode)
+		if err != nil {
+			return nil, err
+		}
+		newNodes = append(newNodes, node)
+	}
+
+	return newNodes, nil
+}
+
+func (cache *overlaycache) findReputableNodesQuery(ctx context.Context, req *filterReputableRequest) (*sql.Rows, error) {
+	auditCount := req.minReputation.AuditCount
+	auditSuccessRatio := req.minReputation.AuditSuccessRatio
+	uptimeCount := req.minReputation.UptimeCount
+	uptimeRatio := req.minReputation.UptimeRatio
+	nodeAmt := req.reputableNodeAmount
+
+	var args []interface{}
+	var rows *sql.Rows
+	var err error
+	var nodeTypeStorage int32 = 2
+
+	if len(req.excluded) == 0 {
+		args = append(args, auditCount, req.newNodeAuditThreshold,
+			auditSuccessRatio, uptimeCount, uptimeRatio,
+			req.freeBandwidth, req.freeDisk, nodeTypeStorage, nodeAmt)
+
+		rows, err = cache.db.Query(cache.db.Rebind(`SELECT overlay_cache_nodes.node_id,
+	overlay_cache_nodes.node_type, overlay_cache_nodes.address, overlay_cache_nodes.free_bandwidth,
+	overlay_cache_nodes.free_disk, overlay_cache_nodes.audit_success_ratio,
+	overlay_cache_nodes.audit_uptime_ratio, overlay_cache_nodes.audit_count,
+	overlay_cache_nodes.audit_success_count, overlay_cache_nodes.uptime_count,
+	overlay_cache_nodes.uptime_success_count
+	FROM overlay_cache_nodes
+	WHERE overlay_cache_nodes.audit_count > ?
+	AND overlay_cache_nodes.audit_count > ?
+	AND overlay_cache_nodes.audit_success_ratio > ?
+	AND overlay_cache_nodes.uptime_count > ?
+	AND overlay_cache_nodes.audit_uptime_ratio > ?
+	AND overlay_cache_nodes.free_bandwidth > ?
+	AND overlay_cache_nodes.free_disk > ?
+	AND overlay_cache_nodes.node_type == ?
+	LIMIT ?
+	`),
+			args...)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		args = append(args, auditCount, req.newNodeAuditThreshold,
+			auditSuccessRatio, uptimeCount, uptimeRatio,
+			req.freeBandwidth, req.freeDisk, req.excluded, nodeTypeStorage, nodeAmt)
+
+		rows, err = cache.db.Query(cache.db.Rebind(`SELECT overlay_cache_nodes.node_id,
+	overlay_cache_nodes.node_type, overlay_cache_nodes.address, overlay_cache_nodes.free_bandwidth,
+	overlay_cache_nodes.free_disk, overlay_cache_nodes.audit_success_ratio,
+	overlay_cache_nodes.audit_uptime_ratio, overlay_cache_nodes.audit_count,
+	overlay_cache_nodes.audit_success_count, overlay_cache_nodes.uptime_count,
+	overlay_cache_nodes.uptime_success_count
+	FROM overlay_cache_nodes
+	WHERE overlay_cache_nodes.audit_count > ?
+	AND overlay_cache_nodes.audit_count > ?
+	AND overlay_cache_nodes.audit_success_ratio > ?
+	AND overlay_cache_nodes.uptime_count > ?
+	AND overlay_cache_nodes.audit_uptime_ratio > ?
+	AND overlay_cache_nodes.free_bandwidth > ?
+	AND overlay_cache_nodes.free_disk > ?
+	AND overlay_cache_nodes.node_id NOT IN (?`+strings.Repeat(", ?", len(req.excluded)-1)+`)
```

Comment:
If we do still need to support excluded, this might be better sent as an array. You can send array arguments directly to pg using `pq.Array()`. IIRC this would look like

```
AND node_id NOT IN ?::text[]
```
and then you'd bind `pq.Array(req.excluded)` for the corresponding argument.

Reply:
cool, thank you!